lib/Onyx.js

```diff
@@ -1414,32 +1436,7 @@ function update(data) {
         }
     });
 
-    const promises = [];
-    let clearPromise = Promise.resolve();
-
-    _.each(data, ({onyxMethod, key, value}) => {
-        switch (onyxMethod) {
-            case METHOD.SET:
-                promises.push(() => set(key, value));
-                break;
-            case METHOD.MERGE:
-                promises.push(() => merge(key, value));
-                break;
-            case METHOD.MERGE_COLLECTION:
-                promises.push(() => mergeCollection(key, value));
-                break;
-            case METHOD.MULTI_SET:
-                promises.push(() => multiSet(value));
-                break;
-            case METHOD.CLEAR:
-                clearPromise = clear();
-                break;
-            default:
-                break;
-        }
-    });
-
-    return clearPromise.then(() => Promise.all(_.map(promises, p => p())));
+    return innerUpdate(data);
```

Comment:
As a data point, [I tried the proposed solution by Marc](https://github.com/Expensify/react-native-onyx/compare/marcaaron-resolveOnBroadcast) which would try to resolve earlier while keeping the writes async, this however is not correct and broke a bunch of other tests.